frontend/public/components/secrets/create-secret.tsx

```diff
@@ -307,11 +628,21 @@ class SSHAuthSubform extends React.Component<SSHAuthSubformProps, SSHAuthSubform
 }
 
 const secretFormFactory = secretType => {
-  return secretType === SecretTypeAbstraction.webhook ? withSecretForm(WebHookSecretForm) : withSecretForm(SourceSecretForm);
+  switch (secretType) {
+    case SecretTypeAbstraction.source:
+      return withSecretForm(SourceSecretForm);
+    case SecretTypeAbstraction.image:
+      return withSecretForm(ImageSecretForm);
+    default:
+      return withSecretForm(WebHookSecretForm);
+  }
 };
 
 const SecretLoadingWrapper = props => {
-  const secretTypeAbstraction = determineSecretTypeAbstraction(_.get(props.obj.data, 'data'));
+  if (!props.obj.loaded) {
+    return <LoadingBox />;
+  }
+  const secretTypeAbstraction = determineSecretTypeAbstraction(_.get(props.obj, 'data.type'));
```

Comment:
Not sure if `_.get` is needed here if we're checking if it's loaded already?

Reply:
yeah, its not necessary here :+1: